#! /bin/sh /usr/share/dpatch/dpatch-run
## fix_i386_gcc4_build.dpatch by Julien Cristau <julien.cristau@ens-lyon.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch from upstream to fix build on i386 with gcc4.
## DP: This is recorded as PR#3604 and PR#3637
## DP: http://camlcvs.inria.fr/cgi-bin/cvsweb/ocaml/otherlibs/num/bng_ia32.c.diff?r1=1.3;r2=1.3.6.1

@DPATCH@
diff -urNad ocaml-3.08.3.orig/otherlibs/num/bng_ia32.c ocaml-3.08.3/otherlibs/num/bng_ia32.c
--- ocaml-3.08.3.orig/otherlibs/num/bng_ia32.c	2003-10-26 10:51:11.000000000 +0100
+++ ocaml-3.08.3/otherlibs/num/bng_ia32.c	2005-06-20 17:34:10.015173470 +0200
@@ -121,8 +121,8 @@
         "leal 4(%1), %1 \n\t"
         "decl %2 \n\t"
         "jnz 1b"
-        : "+&r" (a), "+&r" (b), "+&rm" (blen), "+&r" (out)
-        : "rm" (d)
+        : "+&r" (a), "+&r" (b), "+&r" (blen), "=m" (out)
+        : "m" (d)
         : "eax", "edx");
   }
   if (alen == 0) return out;
@@ -164,8 +164,8 @@
         "leal 4(%1), %1 \n\t"
         "decl %2 \n\t"
         "jnz 1b"
-        : "+&r" (a), "+&r" (b), "+&rm" (blen), "+&rm" (out), "=&r" (tmp)
-        : "rm" (d)
+        : "+&r" (a), "+&r" (b), "=m" (blen), "=m" (out), "=&r" (tmp)
+        : "m" (d)
         : "eax", "edx");
   }
   if (alen == 0) return out;
